File: leetcode/1-two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> a = new HashMap<>();

        for(int i=0; i<nums.length; i++) {
            int complement = target - nums[i];

            if(a.containsKey(complement)) {
                return new int[]{a.get(complement), i};
            }
            a.put(nums[i], i);
        }
        return new int[]{}; 
    }
}

File: leetcode/1-two-sum/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        ans = {}
        for i in range(len(nums)):
            difference = target - nums[i]
            if difference in ans: 
                return [ans[difference], i]
            ans[nums[i]] = i
        return []